src/lib/urls.js

```diff
@@ -30,3 +30,39 @@ export function getCanonicalPath(path) {
   }
   return parts.join('/');
 }
+
+export function copyLinkToClipboard() {
+  if (!('clipboard' in navigator)) {
+    return;
+  }
+  document
+    .querySelector('main')
+    // ToDo: Use `:is(h2, h3, h4, h5, h6)[id]` once support is better.
+    .querySelectorAll('h2[id], h3[id], h4[id], h5[id], h6[id]')
+    .forEach((heading) => {
+      heading.addEventListener('click', (ev) => {
+        // Don't jump when the '#' is clicked.
+        if (ev.target.nodeName === 'A') {
+          ev.preventDefault();
+        }
+        // Only run once.
+        if (heading.dataset.toasted) {
+          return;
+        }
+        try {
+          navigator.clipboard.writeText(
+            `${location.origin}${location.pathname}#${heading.id}`,
+          );
+          const temp = heading.innerHTML;
+          heading.innerHTML += '&nbsp;<small>(📋 Copied)</small>';
+          heading.dataset.toasted = 'toasted';
+          setTimeout(() => {
+            heading.innerHTML = temp;
+            delete heading.dataset.toasted;
+          }, 2000);
+        } catch (err) {
+          console.warn(err.name, err.message);
+        }
```

Comment:
I was thinking we should remove this. I understand the intent but I found the set timeout can look a bit odd if you mouse out or if you use a keyboard to click the hash. You end up with a sort of oddly spaced "(Copied)" string.


https://user-images.githubusercontent.com/1066253/107593356-06fdf200-6bc4-11eb-92f3-a90561052e04.mp4

Reply:
The old WebFu had a toast in the lower left corner:

<img width="324" alt="Screen Shot 2021-02-11 at 16 28 25" src="https://user-images.githubusercontent.com/145676/107658216-337a4280-6c86-11eb-8801-70b341eac1be.png">

I kinda like the proximity of this solution to the user action for added context.

Reply:
I'm supportive of the PR but still not a big fan of the copy UI... We do have a snackbar similar to devsite but it's a little tricky to use. You can see it in Snackbar.js and how it's used in actions.js.